diagnose.py

```
@@ -41,6 +41,18 @@ def run_command(cmd, shell=False):
         )
+        return {
+            "success": result.returncode == 0,
+            "stdout": result.stdout.strip(),
+            "stderr": result.stderr.strip(),
+            "returncode": result.returncode
+        }
+    except FileNotFoundError:
+        return {"success": False, "error": "Command not found", "stdout": "", "stderr": ""}
+    except subprocess.TimeoutExpired:
+        return {"success": False, "error": "Command timed out", "stdout": "", "stderr": ""}
+    except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)